src/main/java/io/confluent/connect/jdbc/source/DataConverter.java

```diff
@@ -175,6 +181,23 @@ private static void addFieldSchema(ResultSetMetaData metadata, int col,
       }
 
       case Types.NUMERIC:
+        if (numericPrecisionMapping) {
+          int precision = metadata.getPrecision(col);
+          if (metadata.getScale(col) == 0 && precision < 20) { // integer
```

Comment:
To make this conversion totally safe, this should be `precision < 19` since INT64 cannot represent the full range of 19 digit precision integers.